dev/io.openliberty.checkpoint/src/io/openliberty/checkpoint/internal/CheckpointImpl.java

```diff
@@ -79,25 +79,18 @@ public CheckpointImpl(ComponentContext cc, @Reference WsLocationAdmin locAdmin)
         this.cc = cc;
         this.criu = criu;
         this.locAdmin = locAdmin;
-        this.checkpointFeatures = "features".equals(cc.getBundleContext().getProperty(BootstrapConstants.CHECKPOINT_PROPERTY_NAME));
-        this.checkpointApplications = "applications".equals(cc.getBundleContext().getProperty(BootstrapConstants.CHECKPOINT_PROPERTY_NAME));
+        String phase = cc.getBundleContext().getProperty(BootstrapConstants.CHECKPOINT_PROPERTY_NAME);
+        this.checkpointAt = phase == null ? null : Phase.getPhase(phase);
```

Comment:
Isn't `phase == null` an error condition? Seems like it'll be silently ignored and server will launch normally. I think the launcher does some verification of the --at-location-internal passed in. Is there some other supported way to arrive at Phase == null  which would ignore an error in phase name (system property?) ?

Reply:
This is the constructor for the service component, which is an immediate component.  So this gets called even when not doing a `checkpoint` action.  I considered adding a NONE enum value for Phase, but thought that would be a strange external for the SPI.  Instead the phase is null when we should have the various events from "features", "applications" etc. be a no-op.